internal/wrappers/cgo/kuzzle/security.go

```diff
@@ -145,221 +131,332 @@ func kuzzle_security_destroy_role(r *C.role) {
+	options := SetQueryOptions(o)
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.DeleteRole(C.GoString(id), options)
+
+	return goToCStringResult(&res, err)
 }
 
-//export kuzzle_security_user_create_credentials
-func kuzzle_security_user_create_credentials(u *C.user, cstrategy *C.char, ccredentials *C.json_object, o *C.query_options) *C.json_result {
-	strategy := C.GoString(cstrategy)
-	credentials := JsonCConvert(ccredentials)
-	res, err := cToGoUser(u).CreateCredentials(strategy, credentials, SetQueryOptions(o))
+//export kuzzle_security_delete_profile
+func kuzzle_security_delete_profile(k *C.kuzzle, id *C.char, o *C.query_options) *C.string_result {
+	options := SetQueryOptions(o)
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.DeleteProfile(C.GoString(id), options)
 
-	return goToCJsonResult(res, err)
+	return goToCStringResult(&res, err)
+}
+
+//export kuzzle_security_delete_user
+func kuzzle_security_delete_user(k *C.kuzzle, id *C.char, o *C.query_options) *C.string_result {
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.DeleteUser(C.GoString(id), SetQueryOptions(o))
+	return goToCStringResult(&res, err)
 }
 
-//export kuzzle_security_user_create_with_credentials
-func kuzzle_security_user_create_with_credentials(u *C.user, ccredentials *C.json_object, o *C.query_options) *C.user_result {
-	credentials, ok := JsonCConvert(ccredentials).(types.Credentials)
-	if !ok {
-		return goToCUserResult(u.kuzzle, nil, types.NewError("Invalid credentials given", 400))
+//export kuzzle_security_delete_credentials
+func kuzzle_security_delete_credentials(k *C.kuzzle, strategy, id *C.char, o *C.query_options) *C.void_result {
+	err := (*kuzzle.Kuzzle)(k.instance).Security.DeleteCredentials(C.GoString(strategy), C.GoString(id), SetQueryOptions(o))
+	return goToCVoidResult(err)
+}
+
+//export kuzzle_security_create_profile
+func kuzzle_security_create_profile(k *C.kuzzle, id, body *C.char, o *C.query_options) *C.profile_result {
+	options := SetQueryOptions(o)
+	res, err := (*kuzzle.Kuzzle)(k.instance).Security.CreateProfile(C.GoString(id), C.GoString(body), options)
+
+	return goToCProfileResult(k, res, err)
+}
+
+//export kuzzle_security_create_role
+func kuzzle_security_create_role(k *C.kuzzle, id, body *C.char, o *C.query_options) *C.role_result {
+	result := (*C.role_result)(C.calloc(1, C.sizeof_role_result))
+	options := SetQueryOptions(o)
+
+	role, err := (*kuzzle.Kuzzle)(k.instance).Security.CreateRole(C.GoString(id), C.GoString(body), options)
+	if err != nil {
+		Set_role_result_error(result, err)
+		return result
 	}
 
-	res, err := cToGoUser(u).CreateWithCredentials(credentials, SetQueryOptions(o))
+	result.role = goToCRole(k, role, nil)
+
+	return result
+}
 
-	return goToCUserResult(u.kuzzle, res, err)
+//export kuzzle_security_create_user
+func kuzzle_security_create_user(k *C.kuzzle, body *C.char, o *C.query_options) *C.json_result {
```

Comment:
We should not use json_result anymore but string_result as we use char* for json now